### ISSUE TITLE
[Feature] Be able to use the auto-generated collection query combined with a filter (argument "by")

### ISSUE THREAD

**Issue by nandotess** [grafbase/grafbase#229]
**Describe the solution you'd like**
Be able to use the auto-generated collection query combined with a filter (argument "by"). The filter could be an `unique` or `enum`.

**Additional context**
Example of schema:

```
type Company @model {
	domain: String! @unique
	name: String!
	status: CompanyStatus! @default(value: ACTIVE)
}

enum CompanyStatus {
	ACTIVE
	DISABLED
}
```

Example of query:
```
export const GET_COMPANIES_QUERY = /* GraphQL */ `
	query GetCompanies(
		$first: Int
		$last: Int
		$before: String
		$after: String
		$status: CompanyStatus
	) {
		companyCollection(
			first: $first
			last: $last
			before: $before
			after: $after
			orderBy: { createdAt: DESC }
			by: { status: $status }
		) {
			pageInfo {
				startCursor
				endCursor
				hasNextPage
				hasPreviousPage
			}
			edges {
				cursor
				node {
					id
					domain
					name
					status
				}
			}
		}
	}
`;
```

If there is a way, today, to load a collection filtered by an `unique` or `enum`, I apologize for opening this ticket.

Thank you!

**Comment by Finistere** [comment]
Hello! Thanks for the feature request. :) That's in the pipe, we're adding filtering capabilities to the collection query. \o/ It should come out in the coming weeks. The API wasn't totally finalized yet, but it should look like the following:
```graphql
companyCollection(
	first: $first
	last: $last
	before: $before
	after: $after
	orderBy: { createdAt: DESC }
        filter: { status: { eq: $status } }
)
```
You would also be able to filter on `name` and `domain`. I'll come back to this issue once released!

**Comment by nandotess** [comment]
That is amazing! Thanks, @Finistere! Looking forward to seeing this implemented, I will be one of the first users. :)

**Comment by nandotess** [comment]
Hi @Finistere. Thanks for your reply. I was just testing this now and I realized this is not working with `enums`.

I have started a new thread at Discord making this question - https://discord.com/channels/890534438151274507/918807414604726332/1133106940726222910

What is the best way to solve this problem now? Should I change this `status/enum` to be an `enabled/Integer` 1 or 0?

Thanks once again.

**Comment by Finistere** [comment]
As discussed in the discord thread, `@search` did support enums but with a bug. ^^" It's fixed in the cloud and will be for the `0.30` release. To summarize, `@search` should be used for filters currently:

```
query {
  companySearch(first: 10, filter: { status: { eq: ACTIVE } }) {
    ...
  }
}
```

Closing the issue. Don't hesitate if I missed something!
